### PR TITLE
creating passwrod reset functionality

### LINKED ISSUE #59: Implementation for Forgot Password Functionality
**Description:**
Implement the frontend for the "Forgot Password" functionality. This feature includes two main parts:
1. Password reset request form.
2. Reset password page.

**Tasks:**
- [x] Create `PasswordResetRequestForm` component.
- [x] Implement form validation for email.
- [x] Handle form submission with Redux.
- [x] Display success and error messages.
- [x] Create `ResetPasswordForm` component.
- [x] Implement form validation for new password.
- [x] Handle reset password submission with Redux.
- [x] Integrate forms into app routing.
- [x] Style the components according to the Figma design.

### Implementation Plan

**1. Create a Password Reset Request Form:**
   - Create a component `PasswordResetRequestForm` where users can input their email to request a password reset.
   - Form should include:
     - Email input field
     - Submit button
   - Form validation for email format.

**2. Handle Form Submission:**
   - Use Redux to manage the state for the password reset request.
   - Create actions and a reducer for handling the request and response states.

**3. Display Success and Error Messages:**
   - Show a success message if the email is sent successfully.
   - Show an error message if there is an issue (e.g., email not found).

**4. Create a Reset Password Page:**
   - Create a component `ResetPasswordForm` where users can reset their password using a token.
   - Page should include:
     - New password input field
     - Confirm new password input field
     - Submit button
   - Form validation for password strength and matching confirmation.

**5. Handle Reset Password Submission:**
   - Use Redux to manage the state for the password reset submission.
   - Create actions and a reducer for handling the password reset response.

**6. Navigation and Routing:**
   - Integrate the forms into the existing app routing.
   - Ensure smooth navigation between the request form and the reset password page.

**7. Styling:**
   - Apply styles to match the Figma design.
   - Ensure the forms are responsive and accessible.

### PR DESCRIPTION
**Pull request by niyobern**
**Summary**
This pull request introduces significant updates to the authentication flow within our application, specifically focusing on the password recovery and reset functionalities. By refining these processes, we aim to provide a more secure and user-friendly experience for our users when they need to recover access to their accounts

**Changes**

- **Password Recovery Initiation:** Implemented an asynchronous thunk action, `requestPasswordReset`, which sends a request to the backend with the user's email. This initiates the password recovery process by generating a recovery token and sending it to the user's email address (this needs a fix).

- **Password Reset:** Added another asynchronous thunk action, `resetPassword`, which allows users to reset their password using the recovery token sent to their email. This action sends the token and the new password to the backend, where the token is verified, and the password is updated.

**Technical Details**

- Both thunk actions utilize [axios](https://github.com/axios/axios) for making HTTP POST requests to the backend.

- Error handling is implemented to return the error response data using `rejectWithValue` in case of request failures, ensuring that the application can gracefully inform the user about any issues encountered during the process.

**Testing**

1. Checkout this branch by running `git switch 59-implementation-for-forgot-password-functionality` in bash terminal (code is expected to work also on Windows)
2. Run `npm run dev` comand in your terminal
3. go to `[/forgot-password](http://localhost:5173/forgot-password)` in your browser (expected to work accross browsers)
4. Enter your email for the account you want to reset password for
5. click on the link sent to your inbox for password reset. It should redirect you to a page like this `http://localhost:5173/reset-password/{token}` where token is a unique key to help you reset your password
6. Fill the form with your new password two times and click submit
7. You will be redirected to login page if reset password is successful or you will see an error message.

**Screenshots**
image
image
image
image

### CODE REVIEW

**Review by ambroisegithub — APPROVED** [review]
It looks good for merge!!!

**Review by 13XAVI — APPROVED** [review]

**Review by bertrandshema — APPROVED** [review]
It is good fir me

File: src/app/slices/authSlices.ts
Comment (wayneleon1) [inline src/app/slices/authSlices.ts:1]: you must follow the rules for creating slices for example : features/product/productslices.ts.
, you are just keep creating for slices into app directory which is not the best practice

**Review by wayneleon1 — DISMISSED** [review]

**Review by dushimeemma — APPROVED** [review]

File: src/app/slices/authSlices.ts
Comment (niyobern) [reply src/app/slices/authSlices.ts:1]: Why do you create a separate directory for redux while app directory was in first place created for that?

Comment (niyobern) [reply src/app/slices/authSlices.ts:1]: What I think can remove ambiguity and make the code structure more organized is keeping all redux code in one place, grouping related feautures together.

Comment (dushimeemma) [inline src/app/slices/authSlices.ts:1]: @wayneleon1 I agree with you.
@niyobern please work on this.

**Review by dushimeemma — CHANGES_REQUESTED** [review]

**Review by dushimeemma — APPROVED** [review]

**Pull request merged**